lib/consoletest.pm

```diff
@@ -16,5 +16,11 @@ sub post_run_hook {
     $self->clear_and_verify_console;
 }
 
+sub post_fail_hook {
+    # Export logs after failure
+    script_run "journalctl --no-pager -b 0 > /tmp/full_journal.log";
```

Comment:
uh, sorry. The test modules are objects in the sense of OOP (object oriented programming). As the base class of every module has a `post_fail_hook` defining a method `post_fail_hook` in a module will not "create" one where there wasn't one in before **but** *override* the one defined in the base class. By writing an explicit `$self->SUPER::post_fail_hook;` it is ensured the baseclass hook is executed. What the test module post_fail_hook writer can decide is if the base class hook should be not called at all or when, e.g. before or after the other code in the hook. Take for example https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/x11test.pm#L12 : Here we call the post_fail_hook of the baseclass to collect useful logs + more in the x11test hook.